frame/support/src/dispatch.rs

```diff
@@ -1320,19 +1323,40 @@ macro_rules! decl_module {
 		{
 			fn on_runtime_upgrade() -> $return {
 				$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!("on_runtime_upgrade"));
-				{ $( $impl )* }
+				let result = { $( $impl )* };
```

Comment:
if user write `return Ok(..);` inside the impl block, I think they would expect to have storage version written.
Maybe it is better to wrap it into a closure or something, no ?

Reply:
Good point